CoreScriptsRoot/ServerCoreScripts/ClientChat/NewChat.lua

```diff
@@ -0,0 +1,415 @@
+
+con = EventFolder.OnSpeakerExtraDataUpdated.OnClientEvent:connect(function(speakerName, data)
+	local speaker = SpeakerDatabase:GetSpeaker(speakerName)
+	if (not speaker) then
+		speaker = SpeakerDatabase:AddSpeaker(speakerName)
+	end
+
+	for k, v in pairs(data) do
+		speaker[k] = v
+	end
+end)
+table.insert(connections, con)
+
+
+ChatWindow.Frame.Resizer.DragStopped:connect(function(x, y)
+	if (ChatWindow.CurrentChatChannel) then
+		ChatWindow.CurrentChatChannel:ReorderChatMessages()
+	end
+end)
+
+
+LocalPlayer.CharacterRemoving:connect(function()
+	GuiParent.Parent = nil
+	LocalPlayer.CharacterAdded:wait()
+	GuiParent.Parent = LocalPlayer:WaitForChild("PlayerGui")
+end)
+
+EventFolder.GetInitDataRequest:FireServer()
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+--///////////// Code to talk to topbar and maintain set/get core backwards compatibility stuff
+--////////////////////////////////////////////////////////////////////////////////////////////
+local Util = {}
+do
+	function Util.Signal()
+		local sig = {}
+
+		local mSignaler = Instance.new('BindableEvent')
+
+		local mArgData = nil
+		local mArgDataCount = nil
+
+		function sig:fire(...)
+			mArgData = {...}
+			mArgDataCount = select('#', ...)
+			mSignaler:Fire()
+		end
+
+		function sig:connect(f)
+			if not f then error("connect(nil)", 2) end
+			return mSignaler.Event:connect(function()
+				f(unpack(mArgData, 1, mArgDataCount))
+			end)
+		end
+
+		function sig:wait()
+			mSignaler.Event:wait()
+			assert(mArgData, "Missing arg data, likely due to :TweenSize/Position corrupting threadrefs.")
+			return unpack(mArgData, 1, mArgDataCount)
+		end
+
+		return sig
+	end
+end
+
+
+
+
+local function SetVisibility(val)
+	ChatWindow.Visible = val
+	moduleApiTable.VisibilityStateChanged:fire(val)
+end
+
+do
+	moduleApiTable.TopbarEnabled = true
+	moduleApiTable.MessageCount = 0
+	
+	function moduleApiTable:ToggleVisibility()
+		SetVisibility(not ChatWindow.Visible)
+	end
+
+	function moduleApiTable:SetVisible(visible)
+		if (ChatWindow.Visible ~= visible) then
+			SetVisibility(visible)
+		end
+	end
+
+	function moduleApiTable:FocusChatBar()
+		ChatBar:CaptureFocus()
+	end
+
+	function moduleApiTable:GetVisibility()
+		return ChatWindow.Visible
+	end
+
+	function moduleApiTable:GetMessageCount()
+		return self.MessageCount
+	end
+
+	function moduleApiTable:TopbarEnabledChanged(enabled)
+		self.TopbarEnabled = enabled
+		self.CoreGuiChanged:fire(Enum.CoreGuiType.Chat, game:GetService("StarterGui"):GetCoreGuiEnabled(Enum.CoreGuiType.Chat))
+	end
+
+	function moduleApiTable:IsFocused(useWasFocused)
+		return ChatBar:IsFocused()
+	end
+
+	moduleApiTable.ChatBarFocusChanged = Util.Signal()
+	moduleApiTable.VisibilityStateChanged = Util.Signal()
+	moduleApiTable.MessagesChanged = Util.Signal()
+
+
+	moduleApiTable.MessagePosted = Util.Signal()
+	moduleApiTable.CoreGuiChanged = Util.Signal()
+	
+	
+	moduleApiTable.eChatMakeSystemMessage = Util.Signal()
```

Comment:
Rather than prefixing `e` here, can we add an `Event` to the end of the variable names?